pixeltable/dataframe.py

```diff
@@ -119,46 +118,116 @@ def finalize_exec(self) -> None:
             self.filter.release()
 
 
+
 class DataFrame:
     def __init__(
             self, tbl: catalog.TableVersion,
-            select_list: Optional[List[exprs.Expr]] = None,
+            select_list:  Optional[List[Tuple[exprs.Expr, Optional[str]]]] = None,
             where_clause: Optional[exprs.Predicate] = None,
             group_by_clause: Optional[List[exprs.Expr]] = None,
             order_by_clause: Optional[List[Tuple[exprs.Expr, bool]]] = None,  # List[(expr, asc)]
             limit: Optional[int] = None,):
         self.tbl = tbl
+
+        # select list logic
+        DataFrame._select_list_check_rep(select_list) # check select list without expansion
         # exprs contain execution state and therefore cannot be shared
-        self.select_list = copy.deepcopy(select_list)  # None: implies all cols
+        select_list = copy.deepcopy(select_list)
+        exps, column_names = DataFrame._normalize_select_list(tbl, select_list)
+        DataFrame._select_list_check_rep(list(zip(exps, column_names))) # check select list after expansion to catch early
+        self._exps = exps
+        self._column_names = column_names
+
+        self.select_list = select_list
+
         self.where_clause = copy.deepcopy(where_clause)
         self.group_by_clause = copy.deepcopy(group_by_clause)
         self.order_by_clause = copy.deepcopy(order_by_clause)
         self.limit_val = limit
 
+    @staticmethod
+    def _select_list_check_rep(
+        select_list: Optional[List[Tuple[exprs.Expr, Optional[str]]]],
+    ) -> None:
+        """Validate basic select list types. 
+        """
+        if select_list is not None: # basic check for valid select list
+            assert len(select_list) > 0
+            for ent in select_list:
+                assert isinstance(ent, tuple)
+                assert len(ent) == 2
+                assert isinstance(ent[0], exprs.Expr)
+                assert ent[1] is None or isinstance(ent[1], str)
+                if isinstance(ent[1], str):
+                    assert is_valid_identifier(ent[1])
+
+    @staticmethod
+    def _normalize_select_list(
+        tbl : catalog.TableVersion, 
+        select_list : Optional[List[Tuple[exprs.Expr, Optional[str]]]],
+    ) -> Tuple[List[exprs.Expr], List[str]]:
+        """
+        Expand select list information with all columns and their names
+        Returns: 
+            tupel consisting of list of expressions, list of names
```

Comment:
tuple
also, should say what is being return (not just the type): a list of select list items and their respective names